src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Post('logout')
+  async logout(@Req() req: TRequest) {
+    try {
+      return await this.authService.logout(req);
+    } catch (error) {
+      if (error.status)
+        throw new HttpException(error.message, error.status);
+      else throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
PullProofAI

Similar to the previous method, the `logout` method also suffers from the same security vulnerability. It exposes potential internal error details to the client. It's recommended to provide a generic error message without leaking sensitive information.

```diff
- throw new HttpException(error.message, error.status);
+ throw new HttpException('An error occurred during logout.', HttpStatus.BAD_REQUEST);
```

#### Performance Improvement

The usage of `try-catch` blocks in both methods (`githubAuthCallback` and `logout`) might impact performance negatively, especially if exceptions are thrown frequently. Consider refactoring the code to minimize the use of try-catch blocks where possible for better performance.

These changes will enhance the security of the application by preventing the exposure of sensitive error details to clients and improve the overall performance by reducing unnecessary try-catch blocks.


<!-- This is an auto-generated comment by PullProofAI -->